quickwit/quickwit-search/src/collector.rs

```diff
@@ -44,17 +44,24 @@ pub(crate) enum SortBy {
         field_name: String,
         order: SortOrder,
     },
+    TwoFastField {
```

Comment:
```suggestion
    TwoFastField {
```

I'm not fond of the approach...
We end up not being able to support `DocId` and score for not much reason at all.

I think we can decouple the concept of 
- "source of single scalar sort value": the SortBy object before this PR
- using a pair of value for sorting.

I suggest we keep the `SortBy` object and rename it as `SortByComponent`. 
It would have the same `sort_order` and `recover_typed_sort_value` as before.

We can then have a `SortBy` object that goes
```
struct SortByPair {
    first: SortByComponent,
    second: Option<SortByComponent>,
}
```

and have it implement the methods with the prototypes you introduced in this PR 
`pub fn sort_orders(&self) -> (SortOrder, SortOrder)`
and
`fn recover_typed_sort_value(
        &self,
        sort_value1: Option<u64>,
        sort_value2: Option<u64>,
    ) -> (Option<SortValue>, Option<SortValue>)`